stdlib/public/Backtracing/FileImageSource.swift

```diff
@@ -20,51 +20,53 @@ import Swift
 
 enum FileImageSourceError: Error {
   case posixError(Int32)
-  case truncatedRead
+  case outOfRangeRead
 }
 
 class FileImageSource: ImageSource {
-  private var fd: Int32
+  private var _mapping: UnsafeRawBufferPointer
 
   public var isMappedImage: Bool { return false }
 
   private var _path: String
   public var path: String? { return _path }
 
-  public lazy var bounds: Bounds? = {
-    let size = lseek(fd, 0, SEEK_END)
-    if size < 0 {
-      return nil
-    }
-    return Bounds(base: 0, size: Size(size))
-  }()
+  public var bounds: Bounds? {
+    return Bounds(base: 0, size: Size(_mapping.count))
+  }
 
   public init(path: String) throws {
     _path = path
-    fd = _swift_open(path, O_RDONLY, 0)
+    let fd = _swift_open(path, O_RDONLY, 0)
     if fd < 0 {
       throw FileImageSourceError.posixError(_swift_get_errno())
     }
+    defer { close(fd) }
+    let size = lseek(fd, 0, SEEK_END)
+    if size < 0 {
+      throw FileImageSourceError.posixError(_swift_get_errno())
+    }
+    let base = mmap(nil, Int(size), PROT_READ, MAP_FILE|MAP_PRIVATE, fd, 0)
+    if base == nil || base! == UnsafeRawPointer(bitPattern: -1)! {
+      throw FileImageSourceError.posixError(_swift_get_errno())
+    }
+    _mapping = UnsafeRawBufferPointer(start: base, count: Int(size))
   }
 
   deinit {
-    close(fd)
+    munmap(UnsafeMutableRawPointer(mutating: _mapping.baseAddress),
+           _mapping.count)
   }
 
   public func fetch<T>(from addr: Address,
```

Comment:
`addr` is actually an offset, yes? Can we give it a less misleading name?

Reply:
It's really only called `addr` because all of the other implementations of the protocol in question actually genuinely are reading from addresses and not just offsets in a file, but since the name isn't a protocol conformance issue, I don't have any particular problem with changing it…

Reply:
Actually, looking at it, the type still needs to be `Address`, so I don't think we really gain much by renaming it. I think I'm inclined to leave the name alone.